clients/csharp/RewindClient.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Drawing;
+using System.Globalization;
+using System.IO;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+
+namespace Com.CodeGame.CodeWars2017.DevKit.CSharpCgdk
+{
+    /// <inheritdoc />
+    /// <summary>
+    ///     A simple implementation of rewind client in C#.
+    /// </summary>
+    /// <remarks>
+    ///     Note: You have to reference System.Drawing in order to use the client.
+    ///     Usage:
+    ///     RewindClient.Instance.AreaDescription(2, 4, AreaType.Cloud);
+    ///     RewindClient.Instance.End();
+    /// </remarks>
+    /// <seealso cref="T:System.IDisposable" />
+    // ReSharper disable once UnusedMember.Global
+    // ReSharper disable once StyleCop.SA1201
+    public class RewindClient : IDisposable
+    {
+        private const int BufferSizeBytes = 1 << 20;
+
+        private const int DefaultLayer = 2;
```

Comment:
I think it would be better replace with 3 by default, as in other clients. Because on client side you send layer 1-5 and it transformed to 0-4 in visualizer. So by default (if layer is not specified) all is drawn on layer 3 from client perspective

Reply:
Updated.
btw, if I send a shape with layer=5 I get the following message in the rewind-viewer console:
`NetListener.cpp:65    WARN| Got message with layer 5, but should be in range 1-5`
But the shape is rendered.

Reply:
Yeah, incorrect check. Fixed it just now